### PR TITLE
make list-view compatible with ember 1.10 and HTMLBars

### LINKED ISSUE #199: Helpers are broken with Ember 1.10.0 with HTMLBars
Since https://github.com/emberjs/ember.js/commit/ad62ccfdfd9578884261c96a2af4a6a238bd3ae9, HTMLBars.registerHelper is private. This broke the helpers here: https://github.com/emberjs/list-view/blob/master/packages/list-view/lib/main.js#L14
Could we use the private method, or use the public makeBoud/ViewHelper method ?

Also unless I'm missing something Ember.HTMLBars.helpers does not exists. https://github.com/emberjs/list-view/blob/master/packages/list-view/lib/helper.js#L22

### PR DESCRIPTION
**Pull request by sly7-7**
Fixes #199 

cc/ @rwjblue

### CODE REVIEW

File: packages/list-view/lib/helper.js [hunk 1]
```diff
@@ -17,9 +17,8 @@ function createHelper(view) {
           delete hash[prop];
         }
       }
-
       /*jshint validthis:true */
-      return Ember.HTMLBars.helpers.collection.helperFunction.call(this, [view], hash, options, env);
+      return env.helpers.collection.helperFunction.call(this, [view], hash, options, env);
```
Comment (rwjblue) [inline packages/list-view/lib/helper.js]: :+1:

File: packages/list-view/lib/main.js [hunk 2]
```diff
@@ -11,5 +11,6 @@ Ember.ListItemView         = ListItemView;
 Ember.ListView             = ListView;
 Ember.ListViewHelper       = ListViewHelper;
 
-(Ember.HTMLBars || Ember.Handlebars).registerHelper('ember-list', EmberList);
-(Ember.HTMLBars || Ember.Handlebars).registerHelper('ember-virtual-list', EmberVirtualList);
+var registerHelper = Ember.HTMLBars._registerHelper || Ember.Handlebars.registerHelper;
```
Comment (rwjblue) [inline packages/list-view/lib/main.js]: Should be:

```javascript
var registerHelper;
if (Ember.HTMLBars) {
  // registerHelper was used for some 1.10-beta's and _registerHelper is for 1.10.0 final.
  registerHelper = Ember.HTMLBars._registerHelper || Ember.HTMLBars.registerHelper;
} else {
  registerHelper = Ember.Handlebars.registerHelper;
}
```

Comment (sly7-7) [inline packages/list-view/lib/main.js]: Thanks :+1:

**Comment by sly7-7** [comment]
oops, sorry, seems like a dup of #197. Though this one seems to be simpler. I must have forgotten some mandatory things.
cc/ @taras

**Comment by sly7-7** [comment]
This is likely superseded by #203

**Pull request closed without merging**